Move building to Docker

oshogbo opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48)
<!-- Reviewable:end -->

oshogbo committed:
Move building to Docker

Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build failed.
https://zuul.gramineproject.io/t/scaffolding/buildset/a0e446d9149a4261988dbc6c32ace4c4

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/dc7f5e2ed7e1492ebd819d8f1c2aafee) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/c09e5705db6a44c3968ff06f14ebda09) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/ec65edb5f47941dba87fadcd53d385b7) SUCCESS in 40s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/a97089bdd9884cb9abc22b22f4e60b43) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/d9ae277cc2ae44379b2224accc0b9b09) SUCCESS in 39s
:x: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/ffc2fedc68094c18a5d7162bac053619) **FAILURE** in 3m 32s
:x: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/93a1bb9f82d94e55b2d7f344fe527ae5) **FAILURE** in 6m 54s
:x: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/455f977ab14f4385a8ea44fd0d1ef5ca) **FAILURE** in 6m 10s
:x: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/d23c771e97844c65bb1f70ea792ffd8d) **FAILURE** in 13m 21s
:x: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/f31ced8ade1d47f8a1f0e84c910e18ef) **FAILURE** in 13m 33s
:x: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/e678acceed814e138ac0a4794a4802a4) **FAILURE** in 24m 32s
:x: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/5eb888da2286405fa1195f31433eb49e) **FAILURE** in 16m 02s
:x: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/ffb20fa1fd354e3c8c0204d220aa3031) **FAILURE** in 15m 33s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/109774508d8b48028051f63fd2946b3d

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/fc92b9d110654f708b70b835b5b88413) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/6bf551a243d24f779a392589d51f018e) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/7579ef96e1834a7eae54dd5f17b3f61c) SUCCESS in 41s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/b5526437faf84c1da1e7c47cf7e04e7b) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/87c1d9bd793f4e71b4deef6f812ba1c1) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/08c153f487b1458f9ea0df23402fd6a1) SUCCESS in 5m 30s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/057f4dacb427472c9e825047a3aa723c) SUCCESS in 8m 53s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/4200f1767c264aaf9e2bd71ca26af72e) SUCCESS in 7m 51s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/0ed09d6fcaf74117ac0c805e89ab360f) SUCCESS in 10m 33s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/77641c09c8db40c785b1cf8eca3444ed) SUCCESS in 13m 40s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/91567bdf05e84682bdaba455dc060f1c) SUCCESS in 11m 55s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/c25e15bf115442788923d7c9fff3bedc) SUCCESS in 14m 30s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/d4b963a6b89f4065a9dfe36b98ea95f0) SUCCESS in 13m 29s

woju reviewed (changes requested):
> Reviewed 24 of 24 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48)* status: all files reviewed, 5 unresolved discussions (waiting on @oshogbo)

___
*[`.zuul.d/playbooks/scaffolding-tests/run.yaml` line 24 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpJwd4MEIH8jphzPQHH:-NpJwd4MEIH8jphzPQHI:bw7wmzd) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/.zuul.d/playbooks/scaffolding-tests/run.yaml#L24)):*
> ```YAML
> 
>     - name: Run hadolint
>       command: ./run-hadolint --ignore DL3008 --ignore DL3009 --ignore DL3059 --ignore DL3006
> ```

Why this is needed?
___
*[`graminescaffolding/builder.py` line 175 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpTzNY4ECm23vpHJAJm:-NpTzNY4ECm23vpHJAJn:b-i8r80h) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/builder.py#L175)):*
> ```Python
>         templates.globals['scag'].update({
>             'builder': self,
>             'MAGIC_DIR': SCAG_MAGIC_DIR,
> ```

Do we need it? Can't we use `{{ scag.builder.project_dir }}`?
___
*[`graminescaffolding/templates/Dockerfile` line 15 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpJtpw3D5VpROnVKnpS:-NpJtpw3D5VpROnVKnpT:ba2n38m) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/templates/Dockerfile#L15)):*
> ```Dockerfile
> {% block install -%}
> RUN apt-get update && \
>     apt-get install --no-install-recommends -y {{ scag.builder.depends | join(' ') }} && \
> ```

This requires `shlex.quote` for each of the arguments, because of `<>` characters used for version specification. We don't use them now, but might in the future.

`{{ scag.builder.depends | map('shquote') | join(' ') }}` I believe? Or something like that.

Or even totally move `depends` into dockerfiles, and expect every framework to add their own `apt-get install` stanza into this `Dockerfile` template. We probably don't need `depends` in python, if `Dockerfile-rootfs` just creates generic docker base image. That way you could also dispose of `{% block preinstall %}`, dotnet can just add two `apt-get install` lines.
___
*[`graminescaffolding/templates/mmdebstrap-hooks/setup.sh` line 5 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpTzH-kClgxJ-b-Eli0:-NpTzH-kClgxJ-b-Eli1:undefined) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/templates/mmdebstrap-hooks/setup.sh#L5)):*
> ```Shell
> set -e
> 
> {% block defualtkeys %}
> ```





_Suggestion:_
```Shell
defaultkeys
```
___
*[`graminescaffolding/templates/frameworks/koajs/Dockerfile` line 3 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpTzs0E-j8lhEgODVfT:-NpTzs0E-j8lhEgODVfU:be71vy) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/templates/frameworks/koajs/Dockerfile#L3)):*
> ```Dockerfile
> {% extends 'Dockerfile' %}
> 
> {% block local %}
> ```

I have afterthoughts about `local`, if is a good thing to propose here. It's confusing because you don't know if it's before or after main `install`. Can we do like in `customize.sh` before, that we have `{% block install %}{{ super() }} and whatever is needed to install after `super()`? (also everywhere else)


<!-- Sent from Reviewable.io -->

gramine-scaffolding commented:
Starting check jobs.

oshogbo reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48)* status: 20 of 24 files reviewed, 5 unresolved discussions (waiting on @woju)

___
*[`.zuul.d/playbooks/scaffolding-tests/run.yaml` line 24 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpJwd4MEIH8jphzPQHH:-NpU8fQo7nujl8MbcoQR:b-fkbuty) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/.zuul.d/playbooks/scaffolding-tests/run.yaml#L24)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Why this is needed?
</blockquote></details>

Because in Dockerfile we do
```
ARG STH
FROM ${STH}
```
And hodlint doesn't like this ;(
___
*[`graminescaffolding/builder.py` line 175 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpTzNY4ECm23vpHJAJm:-NpU8oza42kUnHDhgNL0:bw9k9ac) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/builder.py#L175)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Do we need it? Can't we use `{{ scag.builder.project_dir }}`?
</blockquote></details>

project_dir is the whole path (like `app1/.scag`), and MAGIC_DIR is only `.scag`.
When we use MAGIC_DIR in dockerfile, we are already "chrooted" to app1.
___
*[`graminescaffolding/templates/Dockerfile` line 15 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpJtpw3D5VpROnVKnpS:-NpU96kLCq_w9Ehqfz97:b5x9k66) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/templates/Dockerfile#L15)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

This requires `shlex.quote` for each of the arguments, because of `<>` characters used for version specification. We don't use them now, but might in the future.

`{{ scag.builder.depends | map('shquote') | join(' ') }}` I believe? Or something like that.

Or even totally move `depends` into dockerfiles, and expect every framework to add their own `apt-get install` stanza into this `Dockerfile` template. We probably don't need `depends` in python, if `Dockerfile-rootfs` just creates generic docker base image. That way you could also dispose of `{% block preinstall %}`, dotnet can just add two `apt-get install` lines.
</blockquote></details>

I like that they are in Python because the cleanup section can be done in a single RUN. As hadolint likes it to be. Otherwise, in each Dockerfile, we would need to have this cleanup code.
___
*[`graminescaffolding/templates/mmdebstrap-hooks/setup.sh` line 5 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpTzH-kClgxJ-b-Eli0:-NpU9qnZ8EmpFOJydH2v:bu4ozqt) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/templates/mmdebstrap-hooks/setup.sh#L5)):*
> ```Shell
> set -e
> 
> {% block defualtkeys %}
> ```

Done.
Thanks!
___
*[`graminescaffolding/templates/frameworks/koajs/Dockerfile` line 3 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpTzs0E-j8lhEgODVfT:-NpU9sn0AvGOU9Q4aGyo:b-896fix) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/templates/frameworks/koajs/Dockerfile#L3)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

I have afterthoughts about `local`, if is a good thing to propose here. It's confusing because you don't know if it's before or after main `install`. Can we do like in `customize.sh` before, that we have `{% block install %}{{ super() }} and whatever is needed to install after `super()`? (also everywhere else)
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

gramine-scaffolding commented:
Build failed.
https://zuul.gramineproject.io/t/scaffolding/buildset/6dd705b63e2c4446be999f73c30255a5

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/1bb9b5b82caa4832a19ce33ee2e550a8) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/15d4d9d79cca445ea7433ad71086a49c) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/72221db06daa42a1893d98cc47b8ad56) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/d7af3978b063429a9121d4567fe1d91c) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/d06c444380b64a55a1afe5c217cd929f) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/97b9f1cd3b934252b4aad91a0b021515) SUCCESS in 3m 39s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/c371c755ddb248adb390a19635f52446) SUCCESS in 4m 21s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/d00bad4802104d39ac5d47a7c3ea36d4) SUCCESS in 4m 38s
:x: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/24719397b48e485593ebcbb48cfe7046) **FAILURE** in 4m 54s
:x: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/1eb69763943a49d69b146ea2fcd508be) **FAILURE** in 7m 56s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/bc2850c67c594caf96f87b410305ec52) SUCCESS in 7m 24s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/fb1fa3256e7a4f8c9918d295118683c0) SUCCESS in 9m 26s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/4ec2374d2cc0472788d4830bb2439758) SUCCESS in 9m 29s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/43783b3af9ca4c5b8eac228cfbe83e30

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/9281f8e15c2941e7a2b1b843fabf3332) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/64ec18b93bd6431da0e40e07476e3933) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/034d2680b06d40449decfdb64e474160) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/1aee11fa607b4d8aaacd3a8be6787729) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/a304793db8b24d3e94db76be4798ec50) SUCCESS in 37s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/cbe763f9894c4c188afec588c57c9ff9) SUCCESS in 4m 27s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/ea190bc5e0964d59a4433f88009abf1e) SUCCESS in 4m 27s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/b1e6452528cb4293bd0d682849240510) SUCCESS in 5m 07s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/ec679cad4484404d865220c1792f5b3b) SUCCESS in 11m 02s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/ab023f26799f48ceb51a773070cf23aa) SUCCESS in 10m 02s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/7eb917ee7403491da4eae0f9d4995af0) SUCCESS in 6m 41s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/17a5e19df2344ec19a9f78670727a38b) SUCCESS in 14m 33s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/c9f8bf35911b47f7a02d39b932913843) SUCCESS in 9m 51s

woju committed:
templates/Dockefile: Add macro for RUN apt-get install

Best practice for using apt-get inside Dockerfiles is to apt-get update,
then apt-get install, then clean up caches as a single RUN. Otherwise
you might get into caching problems. This macro will generate correct
command for a given set of packages.

Use this macro to move Builder.depends from Python world to Dockerfiles.

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>
Signed-off-by: Mariusz Zaborski <oshogbo@invisiblethingslab.com>

gramine-scaffolding commented:
Starting check jobs.

woju reviewed (changes requested):
> Reviewed 3 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48)* status: 17 of 28 files reviewed, 2 unresolved discussions (waiting on @oshogbo)

___
*[`.zuul.d/playbooks/scaffolding-tests/run.yaml` line 24 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpJwd4MEIH8jphzPQHH:-NpeajigBliGw4yc-ZJn:b4hf6lv) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/.zuul.d/playbooks/scaffolding-tests/run.yaml#L24)):*
<details><summary><i>Previously, oshogbo (Mariusz Zaborski) wrote…</i></summary><blockquote>

Because in Dockerfile we do
```
ARG STH
FROM ${STH}
```
And hodlint doesn't like this ;(
</blockquote></details>

Ack. Too bad for hadolint.
___
*[`graminescaffolding/builder.py` line 175 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpTzNY4ECm23vpHJAJm:-NpelM2V9tkNmrmiAAyT:bxkez2j) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/builder.py#L175)):*
<details><summary><i>Previously, oshogbo (Mariusz Zaborski) wrote…</i></summary><blockquote>

project_dir is the whole path (like `app1/.scag`), and MAGIC_DIR is only `.scag`.
When we use MAGIC_DIR in dockerfile, we are already "chrooted" to app1.
</blockquote></details>

Ah. So this is relative to docker context, not to cwd from where you run `scag-build` command, right? Can you 1) make it lowercase as everything else (or, less preferably, make everything else uppercase...); 2) document it in somewhere, probably in`hacking.rst`?
___
*[`graminescaffolding/templates/Dockerfile` line 15 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpJtpw3D5VpROnVKnpS:-NpekuNp-97LhyAuTcrC:b-ivrlj5) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/templates/Dockerfile#L15)):*
<details><summary><i>Previously, oshogbo (Mariusz Zaborski) wrote…</i></summary><blockquote>

I like that they are in Python because the cleanup section can be done in a single RUN. As hadolint likes it to be. Otherwise, in each Dockerfile, we would need to have this cleanup code.
</blockquote></details>

We can have it as Jinja macro generating those stanzas, and I don't particularly care about cleanliness of rendered Dockerfiles. I've pushed one version on top of your PR, can you see if you'd accept this?


<!-- Sent from Reviewable.io -->

gramine-scaffolding commented:
Build failed.
https://zuul.gramineproject.io/t/scaffolding/buildset/f82421a5bb8b47cca4a210bf2efadb52

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/7449254cf2024c488397bb07681285c4) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/d76b99067c3a4bfa84db9093454a173e) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/a1c30bc121764d378e54a8b8f9d5acdc) SUCCESS in 39s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/56307b266d884c8684eb9a989bdb63a3) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/f9fd6e87f0b7495db6ee1f07e61daf76) SUCCESS in 38s
:x: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/abf151f6b8f442438e39af1337bd481e) **FAILURE** in 57s
:x: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/92c39bb88086468484d049edbeb8e84e) **FAILURE** in 58s
:x: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/402733ab24a6462484a000286b6062c9) **FAILURE** in 58s
:x: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/285947f144a4489a8eb1054a554a6a49) **FAILURE** in 57s
:x: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/e982fa12c48e41adab5c6c38aa6ed460) **FAILURE** in 58s
:x: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/2ca7d17575104fd0a295ca88d9186672) **FAILURE** in 57s
:x: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/d1cf4f29a34f4c87aac99a0b230f56c4) **FAILURE** in 1m 00s
:x: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/6ca23c8c2ce743f090f88bd189d26799) **FAILURE** in 57s

gramine-scaffolding commented:
Starting check jobs.

oshogbo reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48)* status: 17 of 28 files reviewed, 2 unresolved discussions (waiting on @woju)

___
*[`graminescaffolding/builder.py` line 175 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpTzNY4ECm23vpHJAJm:-NpsgyyxErlXbGKmirTN:b-16i2gs) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/builder.py#L175)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

Ah. So this is relative to docker context, not to cwd from where you run `scag-build` command, right? Can you 1) make it lowercase as everything else (or, less preferably, make everything else uppercase...); 2) document it in somewhere, probably in`hacking.rst`?
</blockquote></details>

1) done.
2) done.
___
*[`graminescaffolding/templates/Dockerfile` line 15 at r1](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NpJtpw3D5VpROnVKnpS:-Npsh-eyBlvClDZ1Jj8q:bwf0ggl) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/9105cedb4b93242b12684fe26219948bcb8b3f04/graminescaffolding/templates/Dockerfile#L15)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

We can have it as Jinja macro generating those stanzas, and I don't particularly care about cleanliness of rendered Dockerfiles. I've pushed one version on top of your PR, can you see if you'd accept this?
</blockquote></details>

Sure :)


<!-- Sent from Reviewable.io -->

gramine-scaffolding commented:
Build failed.
https://zuul.gramineproject.io/t/scaffolding/buildset/4a33406e09f94a5da8665d12f3136099

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/c7b807109e314439a29958b1893aeced) SUCCESS in 38s
:x: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/a5f770fc149c497d809df35ab38caff2) **FAILURE** in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/6661ff1c209e4c22bbdcdc7f676de2bd) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/9afaf1595cc94d0391e274ac106e7b95) SUCCESS in 1m 05s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/c854b3f748ba4b65bb806367c4e4e7dc) SUCCESS in 38s
:x: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/4517f87714d847c08670a96da8c207a7) **FAILURE** in 3m 21s
:x: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/0ec2425b7cc9487087aa9dd8dcf49846) **FAILURE** in 3m 58s
:x: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/28dcaac847da4b78b08577f7cc3909c6) **FAILURE** in 3m 48s
:x: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/fd13d88eb3c24d288476c36898da2e74) **FAILURE** in 8m 03s
:x: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/4cc2a8573d904af8a904a2fe7baf2f4e) **FAILURE** in 9m 28s
:x: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/b0850e0f49374782b0a5c02e40103bc1) **FAILURE** in 6m 16s
:x: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/508d4cc2340648f78b10191a631c4083) **FAILURE** in 9m 27s
:x: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/46e3e9b551914c078c859953ad6fb6e4) **FAILURE** in 7m 58s

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/3b19430d950d4ba7b67ff2cf5fdc021c

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/20aa0548cc9647ef863231a2a4866c2d) SUCCESS in 38s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/69ca734cd2cd4e579eb7bc66623e0f66) SUCCESS in 49s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/e8faecc36ac748b3858023632373d1f9) SUCCESS in 40s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/bad850d5dad8458c95232fb81fca1ef4) SUCCESS in 1m 04s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/3e3586e717ad4706b7559f5c86254b54) SUCCESS in 43s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/1511742dc9d64dffbfe444ffab0c6efd) SUCCESS in 4m 48s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/e94fdb933444448a82c885dd758a16ae) SUCCESS in 5m 51s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/fffa5bedc0b64a62ac34693c2ba401b0) SUCCESS in 4m 18s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/23156707554f44ceb331ac780d755fd4) SUCCESS in 10m 17s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/b98b4019c42c47c4ad3d894e99f6216f) SUCCESS in 12m 17s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/025bc1bc0819456b9b421a5f9e9c1478) SUCCESS in 8m 53s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/bdd918f1229a466c9124d34baae6da52) SUCCESS in 10m 55s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/a7d3e3ccb2754016b0c34d5e90bc9559) SUCCESS in 8m 28s

woju reviewed (changes requested):
> Reviewed 11 of 11 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48)* status: all files reviewed, 1 unresolved discussion (waiting on @oshogbo)

___
*[`Documentation/hacking.rst` line 100 at r8](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NptXixD1e6DQibDpsQ8:-NptXixD1e6DQibDpsQ9:bd9wtwg) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/2d81db2075d314f74abcc41e3e27ffcf50b99f0f/Documentation/hacking.rst#L100)):*
> ```reStructuredText
> ``scag.builder.relative_magic_dir``
>     The path to a directory that contains all files generated during the build
>     phase. This path is relative to the `project_dir`, so it should be used
> ```

This doc and name is incorrect, "relative to the project_dir" means that you can join this path to project_dir to get this path. `scag.magic_dir` is relative to `builder.project_dir`.

If you want to avoid two `magic_dir` variables that would be easy to confuse, I think we can name them `scag.magic_dir` and `builder.scag_dir`.


<!-- Sent from Reviewable.io -->

oshogbo reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48)* status: all files reviewed, 1 unresolved discussion (waiting on @woju)

___
*[`Documentation/hacking.rst` line 100 at r8](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48#-NptXixD1e6DQibDpsQ8:-Nptd-jHAXGOFbWliVXs:b-896fix) ([raw file](https://github.com/intel/ScaffoldingForGramine/blob/2d81db2075d314f74abcc41e3e27ffcf50b99f0f/Documentation/hacking.rst#L100)):*
<details><summary><i>Previously, woju (Wojtek Porczyk) wrote…</i></summary><blockquote>

This doc and name is incorrect, "relative to the project_dir" means that you can join this path to project_dir to get this path. `scag.magic_dir` is relative to `builder.project_dir`.

If you want to avoid two `magic_dir` variables that would be easy to confuse, I think we can name them `scag.magic_dir` and `builder.scag_dir`.
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

gramine-scaffolding commented:
Starting check jobs.

gramine-scaffolding commented:
Build succeeded.
https://zuul.gramineproject.io/t/scaffolding/buildset/81600ba501e34733b25aa585061513b7

:heavy_check_mark: [bandit ](https://zuul.gramineproject.io/t/scaffolding/build/427848098b7f428c8f6548846963b575) SUCCESS in 39s
:heavy_check_mark: [pylint ](https://zuul.gramineproject.io/t/scaffolding/build/76f4f019af534cb981e116cade85ca9b) SUCCESS in 50s
:heavy_check_mark: [pytest ](https://zuul.gramineproject.io/t/scaffolding/build/26d61c1818be4f5996f00b590d2524ac) SUCCESS in 38s
:heavy_check_mark: [build-deb ](https://zuul.gramineproject.io/t/scaffolding/build/82259858f8dd40cc833e41924cdbc71c) SUCCESS in 1m 06s
:heavy_check_mark: [build-docs ](https://zuul.gramineproject.io/t/scaffolding/build/8d46045ddd1d44db9c034c7621dec61a) SUCCESS in 38s
:heavy_check_mark: [test-python ](https://zuul.gramineproject.io/t/scaffolding/build/53d282730eba42e0849a8fc3fe7bf255) SUCCESS in 4m 15s
:heavy_check_mark: [test-flask ](https://zuul.gramineproject.io/t/scaffolding/build/a93330117e70476c9a5a437fda495468) SUCCESS in 5m 39s
:heavy_check_mark: [test-nodejs ](https://zuul.gramineproject.io/t/scaffolding/build/164d6f8e278d44e08a53124fa9d2e351) SUCCESS in 5m 54s
:heavy_check_mark: [test-expressjs ](https://zuul.gramineproject.io/t/scaffolding/build/08dc76d0c3664af7a29c99135898811c) SUCCESS in 10m 10s
:heavy_check_mark: [test-koajs ](https://zuul.gramineproject.io/t/scaffolding/build/e5ba93eed07b4d339f178c124192a972) SUCCESS in 10m 32s
:heavy_check_mark: [test-java-jar ](https://zuul.gramineproject.io/t/scaffolding/build/b0bca3779c6d4ae981e43ee2be063de9) SUCCESS in 6m 25s
:heavy_check_mark: [test-java-gradle ](https://zuul.gramineproject.io/t/scaffolding/build/3e836c3ed81345848a835fb198d5bc65) SUCCESS in 11m 23s
:heavy_check_mark: [test-dotnet ](https://zuul.gramineproject.io/t/scaffolding/build/ca41abb7f3fa4423931fe1b88b4ea9b7) SUCCESS in 9m 25s

woju reviewed (approved):
> Reviewed 11 of 11 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/intel/ScaffoldingForGramine/48)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @oshogbo)



<!-- Sent from Reviewable.io -->
